js/example.js

```
@@ -13,6 +13,10 @@ module.exports = function (styleguide, filePath) {
   const buildPath = styleguide.path.build()
   let data = resolver.data(buildPath, filePath)
 
+  if (Object.keys(data).length === 0) {
```

I don't think this check is necessary?

---

In the case that a developer has created a parseable JSON file and the file contents are `{}`. Maybe not a likely case? But, one that would kill the watch process otherwise.

---

That would be caught by the try/catch block you have in `task/ui.js` though, right?

---

Correct. I was hoping to provide a better error message to the developer. For example, if I let the error bubble up, it reads: `TypeError: Cannot read property 'slice' of undefined at ...` vs `Error: Example data contains zero keys at ...`.

---

In that case, I think it'd be better to throw an error here with the more meaningful message to be captured by try/catch. That way, additional messaging can be centralized in a single place (e.g. appending ` at [${filePath}]!`).

---

👍